### ISSUE TITLE
Rigidbody2D never sleep if they are "characters"

### ISSUE THREAD

**Issue by SuperiorHobo** [godotengine/godot#7996]
Operating system or device - Godot version:
Godot - 2.1 or 2.0/ windows 7

Issue description:
I put some 2d Rigidbodies with "character" mode in scene, they have no any forces applied to them, but debugger show this objects as "active" (with "rigid" mode they sleep and become non active just fine)

Steps to reproduce:
create RigidBody2D with Character mode and zero gravity, watch in monitoring window how it not sleep. Then change mode on Rigid and watch youre have -1 active body

**Comment by Hinsbart** [comment]
Mhh this seems to be meant as a feature, not a bug :P
```cpp
//Body2DSW::sleep_test
else if (mode==Physics2DServer::BODY_MODE_CHARACTER)
	return !active; // characters and kinematic bodies don't sleep unless asked to sleep
```

**Comment by SuperiorHobo** [comment]
why?

**Comment by kubecz3k** [comment]
Yes by design characters should not sleep unless they are explicit asked to, since quite often their logic is processed inside `_integrate_forces`.

**Comment by razcore-art** [comment]
Yeah, this doesn't make any sense to me either. As I understand it, even if `_integrate_forces()` is implemented the physics server still runs collisions and movement on the object unless `custom_integrator` is `true`.

I think `sleeping` should be managed by the server as long as `can_sleep` is on and `custom_integrator` is off as well, but not when `custom_integrator` is on perhaps.

This definitely needs to be documented better.

**Comment by Calinou** [comment]
This was resolved in 4.0 as part of https://github.com/godotengine/godot/pull/48908. The change can't be backported to `3.x` to preserve compatibility with existing projects, but [this caveat is now documented](https://github.com/godotengine/godot/pull/41677)
.
